src/Components/ServiceScene/ServiceScene.tsx

```diff
@@ -397,8 +430,29 @@ export class ServiceScene extends SceneObjectBase<ServiceSceneState> {
   }
 
   private subscribeToLogsQuery() {
-    return this.state.$data?.subscribeToState((newState) => {
+    return this.state.$data?.subscribeToState((newState, prevState) => {
       this.updateLoadingState(newState, TabNames.logs);
+      if (newState.data?.state === LoadingState.Done || newState.data?.state === LoadingState.Streaming) {
+        const resultCount = newState.data.series[0].length;
+        if (resultCount !== this.state.logsCount) {
+          this.setState({
+            logsCount: resultCount,
+          });
+        }
+      }
+    });
+  }
+
+  private subscribeToLogsCountQuery() {
+    return this.state.$logsCount?.subscribeToState((newState) => {
+      if (newState.data?.state === LoadingState.Done) {
+        const value: number | undefined = newState.data.series[0]?.fields?.[1]?.values?.[0];
+        if (value !== undefined) {
```

Comment:
What happens if it is undefined and the `totalLogsCount` state is currently set? Should we reset to 0?

Reply:
If the instant query returns "Done" but does not return a result, I guess that would have to be a no-data state, so yeah we should set 0, but the total count won't get displayed if the logs query returns 0, so hopefully it doesn't matter either way.

Updated!